NickelHook.h

```diff
@@ -63,6 +65,12 @@ struct nh_dlsym {
     bool       optional; // default: false - prevents a failure to resolve the sym from being treated as a fatal error, and sets it to NULL instead
 };
 
+// nh_delete_* deletes the provided path if it exists. It performs some checks
+// to detect and avoid deleting critical system files. The functions return true
+// if the path was removed, or false otherwise.
+__attribute__((visibility("default"))) bool nh_delete_file(const char *path);
+__attribute__((visibility("default"))) bool nh_delete_dir(const char *path);
```

Comment:
I'd add a note to the docs about `nh_delete_dir` only removing the dir if it's empty.

Reply:
Good idea.